controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -252,14 +255,83 @@ func (r *OCSInitializationReconciler) ensureRookCephOperatorConfigExists(initial
 			Namespace: initialData.Namespace,
 		},
 	}
-	err := r.Client.Create(r.ctx, rookCephOperatorConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		r.Log.Error(err, fmt.Sprintf("Failed to create %s configmap", util.RookCephOperatorConfigName))
+
+	csiPluginDefaults := defaults.DaemonPlacements["csi-plugin"]
+	csiProvisionerDefaults := defaults.DaemonPlacements["csi-provisioner"]
+
+	pluginTolerations := r.getCsiPluginTolerations()
+	provisionerTolerations := r.getCsiProvisionerTolerations()
+
+	opResult, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rookCephOperatorConfig, func() error {
+
+		if rookCephOperatorConfig.Data == nil {
+			rookCephOperatorConfig.Data = make(map[string]string)
+		}
+
+		if pluginTolerations != nil {
+			tolerations := append(pluginTolerations, csiPluginDefaults.Tolerations...)
+			tolerationsYAML, err := yaml.Marshal(tolerations)
+			if err != nil {
+				return err
+			}
+			rookCephOperatorConfig.Data["CSI_PLUGIN_TOLERATIONS"] = string(tolerationsYAML)
+			initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified = true
+		} else if pluginTolerations == nil && initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified {
+			delete(rookCephOperatorConfig.Data, "CSI_PLUGIN_TOLERATIONS")
+			initialData.Status.RookCephOperatorConfig.CsiPluginTolerationsModified = false
+		}
+
+		if provisionerTolerations != nil {
+			tolerations := append(provisionerTolerations, csiProvisionerDefaults.Tolerations...)
+			tolerationsYAML, err := yaml.Marshal(tolerations)
+			if err != nil {
+				return err
+			}
+			rookCephOperatorConfig.Data["CSI_PROVISIONER_TOLERATIONS"] = string(tolerationsYAML)
+			initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified = true
+		} else if provisionerTolerations == nil && initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified {
+			delete(rookCephOperatorConfig.Data, "CSI_PROVISIONER_TOLERATIONS")
+			initialData.Status.RookCephOperatorConfig.CsiProvisionerTolerationsModified = false
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to create/update rook-ceph-operator-config configmap")
 		return err
 	}
+	r.Log.Info("Successfully created/updated rook-ceph-operator-config configmap", "OperationResult", opResult)
+
 	return nil
 }
 
+func (r *OCSInitializationReconciler) getCsiPluginTolerations() []corev1.Toleration {
+
+	var tolerations []corev1.Toleration
+
+	for _, sc := range r.clusters.GetStorageClusters() {
+		if val, ok := sc.Spec.Placement["csi-plugin"]; ok {
+			tolerations = append(tolerations, val.Tolerations...)
+		}
+	}
+
+	return tolerations
+}
+
+func (r *OCSInitializationReconciler) getCsiProvisionerTolerations() []corev1.Toleration {
+
+	var tolerations []corev1.Toleration
+
+	for _, sc := range r.clusters.GetStorageClusters() {
+		if val, ok := sc.Spec.Placement["csi-provisioner"]; ok {
+			tolerations = append(tolerations, val.Tolerations...)
+		}
+	}
+
+	return tolerations
+}
```

Comment:
make this as helper function and pass `csi-plugin` and `csi-provisioner` as input?

Reply:
done